### DISCUSSION TITLE
Self-hosted backend?

### DISCUSSION THREAD

**Discussion by scambier** [alangrainger/share-note#31] [Vote for upcoming features]
Would it be possible to open source https://share.note.sx and provide instructions for installation? Even if the notes are encrypted, I'd better rely on a service that _I_ host myself, rather than use a 3rd-party service :)

**Comment by alangrainger** [comment]
edit: There are now multiple self-hosted backend options:

https://docs.note.sx/Running-your-own-server

**Reply by scambier** [reply]
It's not about the risk, it's just that I avoid to rely on 3rd party services whenever possible

**Reply by alangrainger** [reply]
Understood. I have the server implementation docs here, but they are about a week out of date. The plugin is in very active development so infrastructure is changing every day 😬 

https://github.com/alangrainger/obsidian-share/blob/main/docs/Running%20your%20own%20server.md

I will get them up-to-date again in a few weeks when things are more stable.

**Reply by scambier** [reply]
Thanks a lot :)

**Reply by thibaultmol** [reply]
@alangrainger consider updating the link in this comments
https://docs.note.sx/self-hosting

**Reply by huyz** [reply]
I'm having a hard time finding instructions on how to modify `data.json` ever since the server was changed from https://github.com/alangrainger/share-note-self-hosted-backend to https://github.com/note-sx/server

https://docs.note.sx/running-your-own-server is outdated

Anyone know where the latest instructions are?

**Reply by alangrainger** [reply]
Here: https://github.com/note-sx/server?tab=readme-ov-file#change-your-obsidian-plugin-to-point-to-your-server

Edit it in any text editor.

**Reply by huyz** [reply]
@alangrainger Thanks, I got it to mostly work. I simply had other issues that were confusing me during migration.

**Comment by zsgcgrs** [comment]
I hope to have a self-hosted backend docker project. The plug-in is very easy to use. Thanks for the development.

**Comment by oldwinter** [comment]
If there is a version with Self-hosted backend, it will be more powerful than obsidian publish

**Comment by 0xGitGuy** [comment]
One click vercel deployment perhaps?

**Comment by zei-bit** [comment]
Thanks for the plugin @alangrainger! It's very intuitive and I haven't had many issues so far. Currently I'm in the process of self hosting my own files as a side project. Do you have any plans on making the self host docs more comprehensive or expanding upon in more detail each step in the API? 

Edit: I'm especially interested in how the createNote endpoint looks like.

**Comment by ItsNiceCraft** [comment]
I can only agree, I would move from Obsidian Publish right now if the current server implementation would be made fully open source. I don't mind using a public service but especially with my notes it'd be great if they could be stored on my own infrastructure only.

**Comment by webenadam** [comment]
+1 This would make the use of this plugin so much more reliable. a great idea!

**Comment by Write** [comment]
+1, really needs it. Thanks a lot.

**Comment by leelax22** [comment]
Like everyone else, I really want to be self-hosted, but from the developer's point of view, there must be something that I'm sorry to reveal.

It would be great if you could simply configure it to go back in the form of a docker.

It might be a bit psychotic, but whether I can post it on my server or not makes me feel at ease.

Among the other plugins, there is one that simply supports self-hosted as a docker, but I find this plug-in also saves css, which is so pretty on the web that it's attractive.

And it doesn't matter much, but it would be great if toc was added.

Anyway, thank you for making such a good plug-in. It's a little hard to share important documents, but I can use it to share it with my friends simply.

**Comment by 0xGitGuy** [comment]
Would there be any point if the backend code be released as-is even if it's not working as self hosted so others can hack something up with community contribution?

It would be a real pity if more people can't have access to this with greater autonomy. It's an elagant solution.

**Comment by alangrainger** [comment]
You can now self-host your own server using the repo here:

https://github.com/alangrainger/share-note-self-hosted-backend

All features should be functional - let me know if anything is missing.

**Reply by Write** [reply]
Thanks a lot !

**Reply by zsgcgrs** [reply]
This is great news, thank you for your contribution

**Reply by zei-bit** [reply]
Awesome, thank you for this!

**Reply by leelax22** [reply]
I can host my own hosting server via apache2, but how do I connect the shared note plug-in to the server?

![image](https://github.com/user-attachments/assets/6e38caae-c326-444d-a16e-efe578f03bad)

Changed the api default address of the plug-in to my server address (I thought api.note.sx should be changed.)
However, it does not work from 'api/v1/account/get-key' after changing and building the plug-in.

I'm sorry if it's a novice question, but need your help😓

**Reply by alangrainger** [reply]
Hi @leelax22,  you'll need to follow the steps in the plugin setup here:

https://github.com/alangrainger/share-note-self-hosted-backend?tab=readme-ov-file#plugin

Specifically you need to set the `server` variable, as well as the `uid` and `apiKey` values. They should be the same as the values in the server `env.cfg` file:

https://github.com/alangrainger/share-note-self-hosted-backend?tab=readme-ov-file#uid-and-private_key

**Reply by leelax22** [reply]
Thank you for your kindness. 
I'm trying to follow it step by step, but it doesn't work at once. 
I think I've done the web configuration on my vm, turned the sql server, and wrote the configuration, but I can't see the log where the error comes from. 
I'll test it more carefully when I have time and ask for help. 
Thank you again😀

**Reply by alangrainger** [reply]
Make sure you close and open Obsidian after changing the `data.json` file, otherwise Obsidian won't pick up the changes.

**Comment by alangrainger** [comment]
You can now self-host your own server using the repo here:

https://github.com/alangrainger/share-note-self-hosted-backend

I will close this discussion - for any issues relating to the new backend please use the Issues on that repo.
